TitanHide/hooks.cpp

```diff
@@ -602,45 +602,81 @@ static NTSTATUS NTAPI HookNtGetContextThread(
     return ret;
 }
 
-static NTSTATUS NTAPI HookNtSetContextThread(
+static NTSTATUS NTAPI SetContextThreadWithoutDebugRegisters(
     IN HANDLE ThreadHandle,
     IN PCONTEXT Context)
 {
-    ULONG pid = (ULONG)(ULONG_PTR)PsGetCurrentProcessId();
-    KPROCESSOR_MODE PreviousMode = ExGetPreviousMode();
-    bool IsHidden = PreviousMode != KernelMode && Hider::IsHidden(pid, HideNtSetContextThread);
     ULONG OriginalContextFlags = 0;
-    if(IsHidden)
+    CONTEXT contextCopy;

```

Comment:
Sorry for the delay, but I now remember why I didn't implement this. It's undocumented, but the sizeof(CONTEXT) isn't actually the maximum size a user might pass to Get/SetThreadContext. See https://github.com/x64dbg/TitanEngine/blob/x64dbg/TitanEngine/TitanEngine.Debugger.Context.cpp#L987, but I'm unfortunately not sure how to deal with this...

Reply:
Hm, I have two possible solutions in mind. I took a look at what `InitializeContext` does, and it calls a function called `RtlGetExtendedContextLength2` to get the length for its `ContextFlags`.

Solution 1: Go with the maximum size - it appears to be  0x4d0?
Solution 2: The same functions also exists in the kernel, add it to the other undocumented functions and use it.

I can do one of those when I have some spare time, right now the bad fix works for me.